lib/stdlib/src/erl_tar.erl

```diff
@@ -1253,37 +1259,38 @@ parse_numeric(<<First, _/binary>> = Bin) ->
 
 parse_octal(Bin) when is_binary(Bin) ->
     %% skip leading/trailing zero bytes and spaces
-    do_parse_octal(Bin, <<>>).
-do_parse_octal(<<>>, <<>>) ->
-    0;
+    do_parse_octal(Bin, 0).
+
 do_parse_octal(<<>>, Acc) ->
-    case io_lib:fread("~8u", binary:bin_to_list(Acc)) of
-        {error, _} -> throw({error, invalid_tar_checksum});
-        {ok, [Octal], []} -> Octal;
-        {ok, _, _} -> throw({error, invalid_tar_checksum})
-    end;
-do_parse_octal(<<$\s,Rest/binary>>, Acc) ->
+    Acc;
+do_parse_octal(<<$\s, Rest/binary>>, Acc) ->
     do_parse_octal(Rest, Acc);
 do_parse_octal(<<0, Rest/binary>>, Acc) ->
     do_parse_octal(Rest, Acc);
 do_parse_octal(<<C, Rest/binary>>, Acc) ->
-    do_parse_octal(Rest, <<Acc/binary, C>>).
+    Digit = C - $0,
```

Comment:
is it worth instead changing this to be `integer_to_list([C])` to avoid parsing non-integers as integers?

Reply:
No. That would be slower and require a `try`/`catch` and is unnecessary because any non-digits will be caught by the range check.